docs/portal.properties-Reference.md

```diff
@@ -280,3 +281,62 @@ This gene set will add the following in the query box:
 ```
 "BRCA genes" BRCA1: MUT=E1258D; BRCA2: HOMDEL MUT=NONSENSE MUT=NONSTART MUT=NONSTOP MUT=FRAMESHIFT MUT=SPLICE MUT=TRUNC;
 ```
+# Ehcache Settings
+cBioPortal is supported on the backend with Ehcache. The configuration, size, and location of these caches are configurable from within portal.properties through the following properties.
+
+First, select a cache configuration using `ehcache.xml_configuration`. This specifies whether to use a disk-only, heap-only, or hybrid (disk + heap) cache. The default value is `ehcache-mixed.xml` which initializes a hybrid caching system.
+```
+ehcache.xml_configuration=[ehcache-heap-only.xml OR ehcache-mixed.xml OR ehcache-disk-only.xml]
+```
+
+If the cache is configured to use disk resources, users must make a directory available and set it with the `ehcache.persistence_path` property. Ehcache will create seperate directories under the provided path for each cache defined in the ehcache.xml_configuration file. 
+```
+ehcache.persistence_path=[location on the disk filesystem where Ehcache can write the cache to /tmp/]
+```
+
+Cache size must be set for both heap and disk; Ehcache requires disk size to be greater than heap size. Default values are provided. The general repository cache is specified to use 1021MB for heap and 4GB for disk. The static repository cache is specified to use 30MB for heap and 32MB for disk. For installations with increased traffic or data, cache sizes can be increased to further improve performance. 
+```
+ehcache.general_repository_cache.max_bytes_heap=
+ehcache.general_repository_cache.max_bytes_heap_units=[size unit e.g MB, GB, entries]
+ehcache.general_repository_cache.max_bytes_local_disk=
+ehcache.general_repository_cache.max_bytes_local_disk_units=[size unit e.g MB, GB, entries]
+
+ehcache.static_repository_cache_one.max_bytes_heap=
+ehcache.static_repository_cache_one.max_bytes_heap_units=[size unit e.g MB, GB, entries]
+ehcache.static_repository_cache_one.max_bytes_local_disk=
+ehcache.static_repository_cache_one.max_bytes_local_disk_units[size unit e.g MB, GB, entries]
+```
+
+To configure a system where no caching occurs, use ehache-heap-only.xml with minimal heap sizes (0 is not supported, so set max_bytes_heap=1 and max_bytes_heap_units=B for all caches) and set
```

Comment:
It just seems weird to me if I set the option cache_enabled to false, and I still need to configure the max_bytes_heap and max_bytes_heap_units to the minimum. Does that mean if I disable the cache and max_bytes_heap accidentally sets to 10GB, the ehcache will still work?

Reply:
If we are able to do conditional configuration of the ehcache-*.xml file as you suggest below, then we can handle this programatically. When ehcache.cache_enabled is false, it could ignore any of the storage footprint sizes and just configure a 1B heap only cache so that ehcache would not throw an exception on startup.
If we cannot do that, maybe we should just give the deployer a fourth configuration file (ehcache-disabled.xml) file with hardcoded values and tell the user to set ehcache.cache_enabled to false and set ehcache.xml_configuration to "ehcache-disabled.xml" (any other resource settings then are ignored due to hardcoding)

Reply:
this is now not required ... if ehcache.cache_enabled is false, other properties don't need to be set.